airflow/providers/trino/hooks/trino.py

```diff
@@ -255,32 +257,38 @@ def get_pandas_df(
     @overload
     def run(
         self,
-        sql,
+        sql: Union[str, Iterable[str]],
         autocommit: bool = False,
-        parameters: Optional[Tuple] = None,
+        parameters: Optional[Union[Iterable, Mapping]] = None,
         handler: Optional[Callable] = None,
-    ) -> None:
+        split_statements: bool = False,
+        return_last: bool = True,
+    ) -> Optional[Union[Any, List[Any]]]:
         """Execute the statement against Trino. Can be used to create views."""
 
     @overload
     def run(
         self,
-        sql,
+        sql: Union[str, Iterable[str]],
         autocommit: bool = False,
-        parameters: Optional[Tuple] = None,
+        parameters: Optional[Union[Iterable, Mapping]] = None,
         handler: Optional[Callable] = None,
+        split_statements: bool = False,
+        return_last: bool = True,
         hql: str = "",
-    ) -> None:
+    ) -> Optional[Union[Any, List[Any]]]:
         """:sphinx-autoapi-skip:"""
 
     def run(
         self,
-        sql,
+        sql: Union[str, Iterable[str]],
         autocommit: bool = False,
-        parameters: Optional[Tuple] = None,
+        parameters: Optional[Union[Iterable, Mapping]] = None,
         handler: Optional[Callable] = None,
+        split_statements: bool = False,
```

Comment:
Something to consider...
I suggest to set it to True
```suggestion
        split_statements: bool = True,
```
I think in trino this should be default True.
Trino doesn't support multiple statement in single query so we need to split them.
Currently Train Operator works only with a single statement. This will make the operator also work with multiple statements
https://github.com/apache/airflow/blob/52cf5793523d3acab480b1933fb7cb33db1c15c7/airflow/providers/trino/operators/trino.py#L81-L83

Alternatively we can do that in a different PR on the Operator level but I think that for all the DBs that need split statment behavior it would be best to set the default as True for them. It just makes everything easier

WDYT?

Reply:
I'm worried that it'll break backward compatibility.
There is already `split_statements: bool = True` for Databriks and Snowflake.
Also, `DrillOperator` splits SQL by `hook.run(..., split_statements=True)` as it did before

Reply:
NOW. THIS is going to be interesting. Exactly as I anticipated. Using this PR to see how we can look at backwards compatibility... 

Exciting :). I will take a look later today :)

Reply:
In Trino, Presto, Snowflake it doesnt make sense to have it as False.

I think the risk is only teoritical. I cant think of an actual example where this is going to break user code but if we want to be on the safe side (and I guess we should) can we maybe deprecate the default value of false and in next major release change it to True?

My argument here is that if we don't set it as True for these providers - users will have to set it to True in **all** usages.